File: medium/javascript/0334-increasing-triplet-subsequence.js
/**
 * O(n) time and O(1) space
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function (nums) {
  let minNum = Infinity;
  let midNum = Infinity;

  for (let currentNum of nums) {
    if (currentNum > midNum) {
      return true;
    }

    if (currentNum > minNum) {
      midNum = currentNum;
    } else {
      minNum = currentNum;
    }
  }

  return false;
};

console.log(increasingTriplet([1, 5, 2, 6, 3]));

File: medium/go/0334-increasing-triplet-subsequence.go
package main

import "math"

// O(n) and O(1) space
func increasingTriplet(nums []int) bool {
    min := math.MaxInt
	mid := math.MaxInt

	for _, v := range nums {
		if v > mid {
			return true
		}

		if v > min {
			mid = v
		} else {
			min = v
		}
	}

	return false
}